src/main/java/graphql/analysis/NodeVisitorWithTypeTracking.java

```diff
@@ -233,20 +233,10 @@ public TraversalControl visitObjectField(ObjectField node, TraverserContext<Node
         return TraversalControl.CONTINUE;
     }
 
-    private static boolean isChildOfVariableDefinition(TraverserContext<Node> context) {
-        if (context == null) {
-            return false;
-        }
-        if (context.getParentNode() instanceof VariableDefinition) {
-            return true;
-        }
-        return isChildOfVariableDefinition(context.getParentContext());
-    }
-
     @Override
     protected TraversalControl visitValue(Value<?> value, TraverserContext<Node> context) {
-        if (isChildOfVariableDefinition(context)) {
-            return TraversalControl.CONTINUE;
+        if (context.getParentNode() instanceof VariableDefinition) {
```

Comment:
We can do even better: we can directly visitVariableDefinition and abort ( I think)